test/unit/query/builder.js

```diff
@@ -5887,4 +5912,99 @@ describe("QueryBuilder", function() {
       });
     });
   });
+
+  describe('toWhereClauseSQL()/toWhereClauseString()', function() {
+    var builder1 = qb().table('test').where('id', 1).andWhere('name', 'knex');
+    var builder2 = qb().table('test')
```

Comment:
Builder could have also joins, select and with statements etc. to show that approach works also for more complex queries.

Reply:
This is true. But join is a clause of its own, even if it does contain a where clause within itself.

This kind of depends on the outcome of your other comment:

**if feature should be where-clause only**
In this case I think it's perfectly fine to exclude join/having/on as these are seperate clauses, even if they contain where clause within itself.

**if feature should not be where-clause only**
In this case the feature should add support for remaining types, such as join. This might be a bit more complex, though. And I don't think it would be possible to contain this within a single function call, but rather split per clause type.

Reply:
I agree that removing where parts of joins are perfectly fine. 

Adding those complexities will test that those additional parts in query will not mess up fetching the `where clause` part of the main query. So it will test that this feature works with also more complex queries than just `select * from "foo" where ...`.

Also will this work when group by is used in the main query? I think that expression builder stuff should be separate from this feature after all.